tensilelite/Tensile/CustomKernels/Custom_Cijk_Alik_Bljk_HHS_BH_UserArgs_MT128x16x128_MI16x16x1_SN_K1_MIWT2_1_triple_buffer.s

```diff
@@ -0,0 +1,3569 @@
+/* numPrefetchIter=1 */
+/* dataAtIterA=1 numReadsIterA=2 skipReadsIterA=1 readsPerIterA=2 */
+/* dataAtIterB=1 numReadsIterB=2 skipReadsIterB=1 readsPerIterB=1 */
+
+/* iter 7 */
+/*  grEndMfmaIndex:9, lwStartMfmaIndex:9, lwEndMfmaIndex:9  */
+/*  numMfmaForLR:1, syncPlrMfmaIndex:12  */
+/*  mfmaIndex:14  */
+ds_read_b128 v[vgprValuA_X2_I0+0:vgprValuA_X2_I0+0+3], v[vgprLocalReadAddrA] offset:64 // L -> Reg lro=32 swapByteOffset=0 ti=128 vIdx=0 eIdx=0 rIdx=0 oIdx=0 buffer=2 iui=0
+ds_read_b128 v[vgprValuB_X2_I0+0:vgprValuB_X2_I0+0+3], v[vgprLocalReadAddrB] offset:64 // L -> Reg lro=32 swapByteOffset=0 ti=16 vIdx=0 eIdx=0 rIdx=0 oIdx=0 buffer=2 iui=0
+s_waitcnt lgkmcnt(14)                              // wait for prior local read local write old=0, new=14 newLW=9 newLR=5
+v_mfma_f32_16x16x16_f16 acc[0:3], v[vgprValuB_X6_I0+0+2+0:vgprValuB_X6_I0+0+2+0+1], v[vgprValuA_X6_I0+0+2+0:vgprValuA_X6_I0+0+2+0+1], acc[0:3] // left value = acc[0+0:3+0]
+/*  mfmaIndex:15  */
+ds_read_b128 v[vgprValuA_X2_I0+4:vgprValuA_X2_I0+4+3], v[vgprLocalReadAddrA] offset:320 // L -> Reg lro=32 swapByteOffset=0 ti=128 vIdx=0 eIdx=1 rIdx=0 oIdx=0 buffer=2 iui=0
+v_mfma_f32_16x16x16_f16 acc[4:7], v[vgprValuB_X6_I0+0+2+0:vgprValuB_X6_I0+0+2+0+1], v[vgprValuA_X6_I0+4+2+0:vgprValuA_X6_I0+4+2+0+1], acc[4:7] // left value = acc[4+0:7+0]
+/* numPrefetchIter=2 */
+/* dataAtIterA=1 numReadsIterA=2 skipReadsIterA=2 readsPerIterA=2 */
+/* dataAtIterB=1 numReadsIterB=2 skipReadsIterB=2 readsPerIterB=1 */
+
+
+label_last_NLL: 
+
+label_toPGR1_0:
+s_cmp_eq_u32 s[sgprGSU], 1                         // GSU == 1 ?
+s_cbranch_scc0 label_GSU_3                         // branch if GSU != 1
+
+/******************************************/
+/* Opt. NoLoadLoop - Begin                */
+/******************************************/
+s_cmpk_eq_u32 s[sgprBeta], 0x0                     // Beta == 0
+s_cbranch_scc0 label_OptNLL_End                    // Branch if Beta is not zero
+
+s_cmp_eq_u32 s[sgprAlpha], 1.0                     // Alpha == 1.0 ?
+s_cbranch_scc0 label_OptNLL_End                    // branch if alpha != 1
+
+s_and_b32 s74, 127, s[sgprSizeI]                   // s74 = s[sgprSizeI] % 128
+s_add_u32 s75, -0x1, s[sgprNumWorkGroups0]
+s_cmp_ge_u32 s[sgprWorkGroup0], s75                // wg0 >= nwg0-1 ?
+s_cselect_b32 s74, s74, 0                          // set rMT0
+s_cmpk_gt_u32 s74, 0x0                             // rMT0 > 0
+s_cbranch_scc1 label_OptNLL_End                    // jump if edges required
+s_and_b32 s74, 15, s[sgprSizeJ]                    // s74 = s[sgprSizeJ] % 16
+s_add_u32 s75, -0x1, s[sgprNumWorkGroups1]
+s_cmp_ge_u32 s[sgprWorkGroup1], s75                // wg1 >= nwg1-1
+s_cselect_b32 s74, s74, 0                          // set rMT1
+s_cmpk_gt_u32 s74, 0x0                             // rMT1 > 0
+s_cbranch_scc1 label_OptNLL_End                    // jump if edges required
+
+
+/* iter 0 (last unrolled loop) */
```

Comment:
Essentially, we should be able to unify Opt and Ord NoLoadLoop.
(move the above branch after last MFMA in NoLoadLoop)
You do not need to make change for this PR, but it might be good to consider that in your next chance.

Reply:
thanks for the suggestions, will consider next time.